compose/utils.py

```diff
@@ -60,13 +64,21 @@ def split_buffer(stream, splitter=None, decoder=lambda a: a):
             yield item
 
     if buffered:
-        yield decoder(buffered)
+        try:
+            yield decoder(buffered)
+        except ValueError as e:
+            log.error(
+                'Compose tried parsing the following chunk as a JSON object, '
+                'but failed:\n%s' % repr(buffered)
+            )
+            raise StreamParseError(e)
```

Comment:
`Exception` would work. The message is also incorrect though. It's not necessarily `JSON`, it could be anything, depending on what gets passed as `decoder()`.